lib/auth/apiserver.go

```diff
@@ -567,22 +567,25 @@ func (s *APIServer) generateKeyPair(auth ClientI, w http.ResponseWriter, r *http
 }
 
 type generateHostCertReq struct {
-	Key        []byte         `json:"key"`
-	Hostname   string         `json:"hostname"`
-	AuthDomain string         `json:"auth_domain"`
-	Roles      teleport.Roles `json:"roles"`
-	TTL        time.Duration  `json:"ttl"`
+	Key         []byte         `json:"key"`
+	HostID      string         `json:"host_id"`
+	NodeName    string         `json:"node_name"`
```

Comment:
this makes API backwards - incompatible, so you have to address this before merging.